src/preamble.js

```diff
@@ -1067,7 +1067,41 @@ function createWasm() {
     var binary;
     try {
       binary = getBinary();
+#if NODE_CODE_CACHING
+      if (ENVIRONMENT_HAS_NODE) {
+        var v8 = require('v8');
+        // Include the V8 version in the cache name, so that we don't try to
+        // load cached code from another version, which fails silently (it seems
+        // to load ok, but we do actually recompile the binary every time).
+        var cachedCodeFile = '{{{ WASM_BINARY_FILE }}}.' + v8.cachedDataVersionTag() + '.cached';
+        cachedCodeFile = locateFile(cachedCodeFile);
+        var hasCached = nodeFS.existsSync(cachedCodeFile);
+        if (hasCached) {
+#if RUNTIME_LOGGING
+          err('NODE_CODE_CACHING: loading module');
+#endif
+          try {
+            module = v8.deserialize(nodeFS.readFileSync(cachedCodeFile));
+          } catch (e) {
+            err('NODE_CODE_CACHING: failed to deserialize, bad cache file? (please delete ' + cachedCodeFile + ')');
```

Comment:
Presumably we should set hasCached=false here so it gets regenerated below?

Reply:
Ah, very good point! Fixed and added testing.

Reply:
So, when the version doesn't match to you hit this exception?    In which case we don't need to include the version here, and the cached version will seamlessly get updated as new v8 versions are used.

Reply:
It turns out it throws when something bad happens like a corrupt data file. But if it's a valid data file, just for a different v8 version, it does not throw, and then it silently throws out the compiled code and recompiles. So we have to use the version in the name to have a chance of actually getting the speedup, I'm afraid.

Reply:
This seems like a design decision that we might be to raise upstream.  I'm pretty sure this would not have been the intended use case.  However given the current limitations of the API I'm not going to block this change.

Reply:
i.e. one would have through there would be a way to know if/when to re-serialize due to incompatible v8 version being used.